crates/primitives/src/utilities.rs

```diff
@@ -59,3 +111,100 @@ pub mod serde_hex_bytes {
         .map_err(|e| serde::de::Error::custom(e.to_string()))
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::GAS_PER_BLOB;
+
+    // https://github.com/ethereum/go-ethereum/blob/28857080d732857030eda80c69b9ba2c8926f221/consensus/misc/eip4844/eip4844_test.go#L27
+    #[test]
+    fn test_calc_excess_blob_gas() {
+        for t @ &(excess, blobs, expected) in &[
+            // The excess blob gas should not increase from zero if the used blob
+            // slots are below - or equal - to the target.
+            (0, 0, 0),
+            (0, 1, 0),
+            (0, TARGET_BLOB_GAS_PER_BLOCK / GAS_PER_BLOB, 0),
+            // If the target blob gas is exceeded, the excessBlobGas should increase
+            // by however much it was overshot
+            (
+                0,
+                (TARGET_BLOB_GAS_PER_BLOCK / GAS_PER_BLOB) + 1,
+                GAS_PER_BLOB,
+            ),
+            (
+                1,
+                (TARGET_BLOB_GAS_PER_BLOCK / GAS_PER_BLOB) + 1,
+                GAS_PER_BLOB + 1,
+            ),
+            (
+                1,
+                (TARGET_BLOB_GAS_PER_BLOCK / GAS_PER_BLOB) + 2,
+                2 * GAS_PER_BLOB + 1,
+            ),
+            // The excess blob gas should decrease by however much the target was
+            // under-shot, capped at zero.
+            (
+                TARGET_BLOB_GAS_PER_BLOCK,
+                TARGET_BLOB_GAS_PER_BLOCK / GAS_PER_BLOB,
+                TARGET_BLOB_GAS_PER_BLOCK,
+            ),
+            // TODO ?
```

Comment:
todo

Reply:
some test cases taken from Geth are outdated, I'll look out for updates in Geth PRs

Reply:
Updated: 1aea2a4b2bae4736ad76c333862c916ae8bc97f1